fix(apigw_restapi_auth check): add method auth testing

n4ch04 opened the pull request:
### Context

Solves #3082 


### Description

When there is no authorized configured at API level, test if all the methods are authorized


### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jfagoagas commented on prowler/providers/aws/services/apigateway/apigateway_restapi_authorizers_enabled/apigateway_restapi_authorizers_enabled.py:
```diff
@@ -13,12 +13,28 @@ def execute(self):
-                report.status = "FAIL"
-                report.status_extended = f"API Gateway {rest_api.name} ID {rest_api.id} does not have an authorizer configured."
+                # we want to know if api has not authorizers and all the resources don't have methods configured
+                api_resources_with_all_methods_authorized = False
+                for resource in rest_api.resources:
+                    # if the resource has methods test if they have all configured authorizer
+                    if resource.resource_methods:
+                        if "NONE" not in resource.resource_methods.values():
+                            api_resources_with_all_methods_authorized = True
+                        else:
+                            # with only one unauthorized method -> return FAIL
+                            api_resources_with_all_methods_authorized = False
+                            break
```
Can you move this logic to the APIGateway service?

n4ch04 replied:
In this case applies to the check

jfagoagas commented on prowler/providers/aws/services/apigateway/apigateway_service.py:
```diff
@@ -95,7 +100,46 @@ def __get_stages__(self):
+                        for resource_method in resource.get(
+                            "resourceMethods", {}
+                        ).keys():
+                            # rest_api.resources.resource_methods.update({resource_method:resource_config.get("authorizationType", None)})
```
```suggestion
```

n4ch04 replied:
Done

jfagoagas commented on prowler/providers/aws/services/apigateway/apigateway_service.py:
```diff
@@ -107,6 +151,11 @@ class Stage(BaseModel):
 
+class PathResourceMethods(BaseModel):
+    path: str
+    resource_methods: dict
```
Can we create a type for this attribute?

n4ch04 replied:
In this case since we are only getting that info adding a type for that info complicates more the check's logic.
In the future if we need to retrieve more info from the resources we will include the model

Fennerr commented:
I have added a [pull request](https://github.com/prowler-cloud/prowler/pull/3188) with some modifications.

I ran this against an account and the logic is working as intended, but the status message could have more information for the end user

n4ch04 commented:
I've added more detail to the `status_extended` field and handled `OPTIONS` case in the service, also added more tests.
Please @Fennerr  check it out.  It was easier modifying the original than working from your PR but thanks for all the help and your work!!

jfagoagas reviewed (approved):
Good one 🚀
